### ISSUE TITLE
Thrown back to play screen with no error when subtitle would occur with directplay and PGS subtitles and the subtitles are off

### ISSUE THREAD

**Issue by sarnex** [jellyfin-archive/jellyfin-roku-legacy#1569]
## Software Versions

- Jellyfin Server Version:  10.8.13
- Roku Client Version:  Problem happens on both 1.6.6 and current unstable branch https://github.com/jellyfin/jellyfin-roku/commit/c9cf7007e454699873bc1d028733a3c27aba82e4

## Describe the bug
This is a really weird one. If I play a video that directplays and the video has PGS subtitles **and the subtitles are off!!!** i get thrown back to the play screen right before a subtitle would appear if they were enabled without an error message. this does not happen if I turn on the subtitles, and does not happen if I force it to transcode for another reason. The web player works fine, so it seems to be a roku client issue. see the video of the issue here: 
https://www.youtube.com/watch?v=No2NBoSAqR8
The problem happens at 0:35.
Just so say again so it's clear, the issue only happens with subtitles **OFF**. If they are on it works fine.
The problem happens with other videos meeting those conditions too, not just the one in the youtube video. A really weird thing is that if I use mkvtoolnix to remux the original mkv file with all the same tracks including the subtitle tracks, that works fine. So it seems it's something with the mkv file, but it plays fine both on the web player and locally on mpv outside of roku.

## How To Reproduce

1. Play video that has PGS subs and directplays and make sure subs are off
2. Wait for a time in the video to occur where there would be a subtitle if they were on
3. You will get thrown to the play screen

## Expected behavior
I expect it to play the video without subtitles

## Logs
telnet log from unstable when subs are off and the problem happens:  https://gist.github.com/sarnex/3fd6dda3587384346265d1f2f7da76d9

jellyfin log from when subs are off and the problem happens: https://gist.github.com/sarnex/d7f028a71de6adf007892bca7b1f07a9

jellyfin and ffmpeg log when subs are on and the problem does not happen (it transcodes): https://gist.github.com/sarnex/2e0bb3c0833a05a2c60c62aae0e153f3 

## Screenshots
Video: https://www.youtube.com/watch?v=No2NBoSAqR8

## Connection Information

- Is server local or remote? Local
- Is server connection HTTP or HTTPS? HTTP

**Comment by 1hitsong** [comment]
Thank you for the detailed bug report! We're looking into the subtitles issues.

FYI for anyone experiencing this issue: We can't update the app till January 2nd due to the [roku holiday blackout](https://blog.roku.com/developer/channel-store-blackouts-2023). **_If_** we're able to fix it before then and push an update release on the 2nd, the app would hit Roku devices a few days later.

Unfortunately, the only workaround until then is to sideload v1.6.6. 

You can find the app here: https://github.com/jellyfin/jellyfin-roku/releases/tag/v1.6.6

You can find instructions here: https://github.com/jellyfin/jellyfin-roku/discussions/897

**Comment by sarnex** [comment]
@1hitsong Thanks for the response! Sorry if I misunderstood your comment, but this issue is still reproducible in both v1.6.6 and the current development branch as of https://github.com/jellyfin/jellyfin-roku/commit/c9cf7007e454699873bc1d028733a3c27aba82e4, so sideloading v1.6.6 would not fix this issue.

**Comment by juanitogan** [comment]
Similarly, with server 10.10.5 and client 2.2.5, if a PGS sub is set to default, it displayed the first title at 3 seconds in (off at 5 seconds) but then bumps you out at 8 seconds (second title not until 48 seconds).  The only reason I suspected the PGS subs is that while testing this repeatedly, when I opened another title afterwards that should have direct played, I found it was transcoding the subs because it said the PGS subs were not supported... except that title has only SRT subs.  Restart the app and retry that second title and it direct plays fine.  Also, the first title plays fine in Roku Media Player, PGS subs and all.

Thus, back to the first title, I remuxed it, replacing the PGS subs with SRT subs, and it now plays fine.  The first title never indicated it was transcoding the subs but, if it was without saying so, then I can assume it was getting the boot because I have the Transcode Killer plugin active.

**Comment by Insprill** [comment]
I'm also running into this issue with Jellyfin v10.10.6 and jellyfin-roku v2.3.0. I'm not sure what information would be helpful to resolve this, but I'm happy to provide anything needed.

**Comment by jimdogx** [comment]
> I'm also running into this issue with Jellyfin v10.10.6 and jellyfin-roku v2.3.0. I'm not sure what information would be helpful to resolve this, but I'm happy to provide anything needed.

We have added some better error logging since this was first reported.  If you could get some Roku logs that would help:

https://github.com/jellyfin/jellyfin-roku/blob/master/docs/DEVGUIDE.md#bugcrash-reports

**Comment by sarnex** [comment]
I just bought a Fire TV Stick and that works fine

**Comment by Insprill** [comment]
Here's the log with subtitles disabled, only transcoding audio. It crashes about a minute in. https://paste.insprill.net/nK4bo
<details>
<summary>Media Info</summary>
<img src="https://github.com/user-attachments/assets/c63869de-bd4b-4ef7-99bc-679ca1553f10"/>
</details>


Here's also a log from a different piece of media with PGS subtitles that crashes before playback even starts. This looks like a different error, so I'm not sure if it's related. If not, let me know and I'll create a new issue unless there's one already open:  https://paste.insprill.net/dAt1C
<details>
<summary>Media Info</summary>
<img src="https://github.com/user-attachments/assets/64a0d898-217e-4dd7-b49b-9998f052e9de"/>
</details>

**Comment by Insprill** [comment]
It seems this issue (or just my issue?) may be related to or a duplicate of #1554. I'm using a Roku 3600X.
